main.go

```diff
@@ -32,8 +33,25 @@ import (
 	"github.com/spf13/cast"
 
 	"github.com/bank-vaults/secret-init/provider"
+	"github.com/bank-vaults/secret-init/provider/file"
 )
 
+func NewProvider(providerName string) (provider.Provider, error) {
+	switch providerName {
+	case file.ProviderName:
+		provider, err := file.NewFileProvider(os.Getenv("SECRETS_FILE_PATH"))
+		if err != nil {
+
+			return nil, err
+		}
+
+		return provider, nil
+	default:
+
+		return nil, errors.New("invalid provider specified")
+	}
```

Comment:
Nit: It's a bit more readable when cases instead of returns are separated by an extra line.
```suggestion
	case file.ProviderName:
		provider, err := file.NewFileProvider(os.Getenv("SECRETS_FILE_PATH"))
		if err != nil {
			return nil, err
		}
		return provider, nil
		
	default:
		return nil, errors.New("invalid provider specified")
	}
```